src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.js

```diff
@@ -0,0 +1,16 @@
+import isValidUrl from '@gdbots/common/isValidUrl';
+
+// supports list of valid Post Urls - https://developers.facebook.com/docs/plugins/oembed
+const getFacebookPostUrl = (str) => {
+  try {
+    const regex = new RegExp('(?:(?:https?:)?\\/\\/)?(?:www\\.)?facebook\\.com\\/([^\\/]+\\/photo(\\.php|s|)(\\/|)|photo(\\.php|s|)(\\/|)|permalink\\.php(\\/|)|media\\/set(\\/|)|questions|notes|[^\\/]+\\/(activity|posts))[\\/?](set=|fbid=|[^\\/]+\\/[^\\/]+\\/|(?=.*\\bstory_fbid=\\b)(?=.*\\bid=\\b).+)?(?:[a-z0-9\\.]+\\/)?([0-9]+)\\/?(?:\\?.[^\\s"]*)?');
+    const postUrlMatches = decodeURIComponent(str).match(regex);
+    return postUrlMatches && isValidUrl(postUrlMatches[0]) ? postUrlMatches[0] : null;
+  } catch (e) {
+    // possible URIError ("malformed URI sequence") exception when used wrongly
```

Comment:
@hunyoboy do you have an example of something that would throw here?

Reply:
@efjacobson also on test! 

https://github.com/triniti/cms-js/blob/c9c808cbbfae86101deb0c97a5498d0691a70f32/src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.test.js#L32

https://github.com/triniti/cms-js/blob/c9c808cbbfae86101deb0c97a5498d0691a70f32/src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.test.js#L29

Reply:
nice